qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -59,14 +72,20 @@ def run(self, dag):
         runs = dag.collect_1q_runs()
         identity_matrix = np.eye(2)
         for run in runs:
-            # Don't try to optimize a single 1q gate
+            single_u3 = False
+            # Don't try to optimize a single 1q gate, except for U3
             if len(run) <= 1:
                 params = run[0].op.params
                 # Remove single identity gates
                 if len(params) > 0 and np.array_equal(run[0].op.to_matrix(),
                                                       identity_matrix):
                     dag.remove_op_node(run[0])
-                continue
+                    continue
+                if (isinstance(run[0].op, U3Gate) and not isinstance(params[0], ParameterExpression)
```

Comment:
Actually I put this in there because I got several test errors where there were ParameterExpressions. If you want me to dig, I can figure out which tests those were. Or we can leave it as is since it works.